In [11]:
from sklearn.preprocessing import MinMaxScaler
import pandas as pd

# Raw data: Age vs. Yearly Income
data = pd.DataFrame({
    'age': [0, 50, 100],
    'income': [20000, 50000, 100000]
})
data

age  income
0    0   20000
1   50   50000
2  100  100000

In [12]:
scaler = MinMaxScaler()

normalized_data = scaler.fit_transform(data)
pd.DataFrame(normalized_data)

0      1
0  0.0  0.000
1  0.5  0.375
2  1.0  1.000

- Minimum value =  0
- Max value = 1

In [13]:
from sklearn.preprocessing import StandardScaler

std_scaler = StandardScaler()

standardized_data = std_scaler.fit_transform(data)
pd.DataFrame(standardized_data)

0         1
0 -1.224745 -1.111168
1  0.000000 -0.202031
2  1.224745  1.313198

- Minimum value = -1.111168
- Max value = 1.313198

- whole data has a mean of 0
- variance of 1

In [ ]:
# Feature creation
# Feature Scaling

data['new_income'] = (data['income'] / 1000).astype(int)
data

age  income  new_income
0    0   20000          20
1   50   50000          50
2  100  100000         100

#### Feature Selection

- Problems, 
- Which features are the most important for our model
- While going through the classes
